### ISSUE TITLE
[Homework] HW6 Retiree Inner 

### ISSUE THREAD

**Issue by Jaegoomon** [kaist-cp/cs492-concur#283]
Hi.

As far as I understand, in `Retirees.inner` threads access the `Vec<>` to free hazard pointer. So I think that I have to control accesses atomically because two more thread can access `inner: Vec<>` concurrently. But is it possible to control atomically  Vector type? Do I have to make my own api such as cas operation or something?

And also, I wonder if the atomic `fence` is really needed. Is it necessary to implement to this project?

**Comment by tomtomjhj** [comment]
> two more thread can access inner: Vec<> concurrently

No. Each thread has its own `Retirees` (note `thread_local!` in `mod.rs`), and all of its method take `&mut self`.

> I wonder if the atomic fence is really needed. 

Yes

**Comment by tomtomjhj** [comment]
> I wonder if the atomic fence is really needed.

This is briefly mentioned in the lecture but I think the explanation wasn't enough, so I'll post more detailed explanations soon.
